tests/functional/src/main/java/org/mule/functional/junit4/rules/ExpectedError.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.functional.junit4.rules;
+
+import static java.lang.String.format;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.fail;
+import static org.mule.tck.junit4.matcher.ErrorTypeMatcher.errorType;
+import org.mule.runtime.api.message.Error;
+import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.exception.MessagingException;
+import org.mule.tck.junit4.matcher.ErrorTypeMatcher;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Throwables;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hamcrest.Matcher;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * JUnit rule for Mule errors
+ *
+ * @since 4.0
+ */
+public class ExpectedError implements TestRule {
+
+  private Matcher<String> messageMatcher;
+  private ErrorTypeMatcher errorTypeMatcher;
+  private Matcher<? extends Throwable> causeMatcher;
+  private List<Matcher> matchers;
+
+  /**
+   * @return a Rule that expects no error to be thrown (identical to behavior without this Rule)
+   */
+  public static ExpectedError none() {
+    return new ExpectedError();
+  }
+
+  /**
+   * Helper method to configure all the matchers that can be used with this rule at the same time.
+   */
+  public static void expectError(ExpectedError error, String namespace, String errorTypeDefinition, Class<?> cause,
```

Comment:
why is this a static method that receives an ExpectedError instead of an instance method in ExpectedError itself?